subsys/bluetooth/host/hci_core.c

```diff
@@ -5464,13 +5466,81 @@ int bt_id_delete(u8_t id)
 	return 0;
 }
 
+#if defined(CONFIG_BT_PRIVACY)
+static void bt_read_identity_root(u8_t *ir)
+{
+#if defined(CONFIG_BT_HCI_VS_EXT)
+	struct bt_hci_rp_vs_read_key_hierarchy_roots *rp;
+	struct net_buf *rsp;
+	int err;
+#endif /* defined(CONFIG_BT_HCI_VS_EXT) */
```

Comment:
For the record, I know I've been trying to stick to the rule of having variable declarations in the beginning of a scope, however that does not seem to be a requirement for Zephyr. There's e.g. many places in the core kernel code where the variable declarations are closer to the point where they're used. If you moved these after the memset you'd only need to have a single ifdef section.

Reply:
Thank you for the clarification. I have moved them to reduce the ifdefs